packages/neo-one-client-core/src/__tests__/Client.test.ts

```diff
@@ -0,0 +1,177 @@
+// tslint:disable no-object-mutation
+import { RawAction } from '@neo-one/client-common';
+import { AsyncIterableX } from '@reactivex/ix-es2015-cjs/asynciterable';
+import { of, of as _of } from 'rxjs';
+import { factory, keys } from '../__data__';
+import { Client } from '../Client';
+import { DeleteUserAccountUnsupportedError, UpdateUserAccountUnsupportedError } from '../errors';
+import { UnlockedWallet } from '../user';
+import { KeyStore, LocalUserAccountProvider, Provider } from '../user/LocalUserAccountProvider';
+const createStoreAndProvider = (wallet: UnlockedWallet) => {
+  const accounts = [wallet.userAccount];
+  const getCurrentUserAccount = jest.fn(() => wallet.userAccount);
+  const getUserAccounts = jest.fn(() => accounts);
+  const selectUserAccount = jest.fn();
+  const deleteUserAccount = jest.fn();
+  const updateUserAccountName = jest.fn();
+  const sign = jest.fn();
+  const keystore: Modifiable<KeyStore> = {
+    currentUserAccount$: _of(wallet.userAccount),
+    getCurrentUserAccount,
+    userAccounts$: _of(accounts),
+    getUserAccounts,
+    selectUserAccount,
+    deleteUserAccount,
+    updateUserAccountName,
+    sign,
+  };
+
+  const network = wallet.userAccount.id.network;
+  const networks = [network];
+  const getNetworks = jest.fn(() => networks);
+  const getUnclaimed = jest.fn();
+  const getOutput = jest.fn();
+  const getTransaction = jest.fn();
+  const getUnspentOutputs = jest.fn();
+  const relayTransaction = jest.fn();
+  const getTransactionReceipt = jest.fn();
+  const getInvocationData = jest.fn();
+  const testInvoke = jest.fn();
+  const getBlockCount = jest.fn();
+  const call = jest.fn();
+  const iterBlocks = jest.fn();
+  const getAccount = jest.fn();
+  const iterActionsRaw = jest.fn();
+  const dataProvider: Modifiable<Provider> = {
+    networks$: _of(networks),
+    getNetworks,
+    getUnclaimed,
+    getOutput,
+    getTransaction,
+    getUnspentOutputs,
+    relayTransaction,
+    getTransactionReceipt,
+    getInvocationData,
+    testInvoke,
+    getBlockCount,
+    getAccount,
+    iterBlocks,
+    iterActionsRaw,
+    call,
+  };
+
+  return {
+    keystore,
+    provider: dataProvider,
+  };
+};
+
+describe('Client Tests', () => {
+  const walletOne = factory.createUnlockedWallet();
+  const walletTwo = factory.createOtherWallet({
+    userAccount: factory.createUserAccount({
+      id: factory.createUserAccountID({
+        address: keys[1].address,
+        network: 'test',
+      }),
+    }),
+  });
+
+  let providerOne: Modifiable<LocalUserAccountProvider<KeyStore, Provider>>;
+  let providerTwo: Modifiable<LocalUserAccountProvider<KeyStore, Provider>>;
+  let client: Client;
+  beforeEach(() => {
+    providerOne = new LocalUserAccountProvider(createStoreAndProvider(walletOne));
```

Comment:
Generally you want to mock the inputs to the class/function under test. Change this to just mock `UserAccountProvider` directly. Look at the neo-one-client-full-core Client.test.ts file for an example.